janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexFilterOptimizerStrategy.java

```diff
@@ -88,9 +88,9 @@ public void apply(final Traversal.Admin<?, ?> traversal) {
                                 && (direction == Direction.BOTH || direction.equals(vertexStep.getDirection().opposite()))) {
                             //Now replace the step with a has condition
                             TraversalHelper.replaceStep(originalStep,
-                                    new HasStep(traversal,
-                                            HasContainer.makeHasContainers(ImplicitKey.ADJACENT_ID.name(), P.eq(vertex))),
```

Comment:
I didn't do enough digging so this is just a question, `.makeHasContainers`  has a return type of  HasContainer[] whereas now I think we'll be passing a single instance. The message `@deprecated As of release 3.2.4. Providers should handle composite {@link P#and} predicates and not rely on splitting.` pretty much answers my question, but I figured I would verify that we're not losing any details.

Reply:
You're right, `makeHasContainers` [consists of this](https://github.com/apache/tinkerpop/blob/3.3.5/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/util/HasContainer.java#L191):

```java
if (predicate instanceof AndP) {
    // not important, as our predicate is P.eq() and therefore not AndP
} else {
    return new HasContainer[]{new HasContainer(key, predicate)};
}
```

So, it just returns a new `HasContainer` wrapped in an array. This array is then provided as a 2nd argument to:

```java
HasStep(final Traversal.Admin traversal, final HasContainer... hasContainers)
```

where it's [used like this](https://github.com/apache/tinkerpop/blob/3.3.5/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/filter/HasStep.java#L45):

```java
Collections.addAll(this.hasContainers, hasContainers);
```

With my change, `hasContainers` will be just one container, but that shouldn't change anything for `Collections.addAll()`. At least, that's my understanding of how varargs should work in Java (but I'm also not really a Java dev).

Reply:
Yes, 
```java
new HasContainer[]{new HasContainer(key, predicate)}
``` 
and 
```java
new HasContainer(key, predicate)
```
will both be the same array in varargs

Reply:
@chupman: Are you ok with these answers?

Reply:
yes